api/v1alpha1/tinkerbell_type.go

```diff
@@ -0,0 +1,132 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=tinkerbell,scope=Namespaced,categories=tinkerbell,singular=tinkerbell,shortName=tb
+// +kubebuilder:storageversion
+
+// Tinkerbell represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
+// Tinkerbell operator watches for different resources such as deployment, services, serviceAccounts, etc. One of those
+// CRs is Tinkerbell which the operator will install the tink-stack based on its specs. Once the CR is deleted,
+// the operator will delete all tinkerbell resources.
+type Tinkerbell struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec describes the desired tinkerbell stack state.
+	Spec TinkerbellStackSpec `json:"spec,omitempty"`
+}
+
+// TinkerbellStackSpec specifies details of an Tinkerbell setup.
+type TinkerbellStackSpec struct {
+	// Version is the Tinkerbell CRD version.
+	Version string `json:"version"`
+
+	// DNSResolverIP is indicative of the resolver IP utilized for setting up the nginx server responsible for proxying
+	// to the Tinkerbell service and serving the Hook artifacts.
+	DNSResolverIP string `json:"clusterDNS"`
```

Comment:
did you still want the json struct tag to be `clusterDNS`?

Reply:
Oh sorry, overlooked that one.

Reply:
Done